common-rendering/src/components/header.tsx

```diff
@@ -0,0 +1,244 @@
+	width: 100%;
+	padding: ${space[1]}px ${space[2]}px ${space[5]}px;
+
+	${from.mobileLandscape} {
+		padding: ${space[1]}px ${space[5]}px ${space[5]}px;
+	}
+
+	${from.desktop} {
+		padding: ${space[1]}px 0 ${space[5]}px;
+	}
+`;
+
+const titleStyles: SerializedStyles = css`
+	display: block;
+	${headline.small()};
+	color: ${text.ctaPrimary};
+`;
+
+const getCreditStyles = (creditColour: string): SerializedStyles => css`
+	display: block;
+	${headline.xxxsmall()};
+	color: ${creditColour};
+	margin-bottom: ${space[1]}px;
+
+	${from.desktop} {
+		position: absolute;
+		left: ${space[5]}px;
+		max-width: 180px;
+	}
+`;
+
+const getBackgroundBottom = (backgroundBottom: string, accentColour: string): SerializedStyles => css`
+	display: flex;
+	padding: ${space[1]}px ${space[2]}px;
+	background-color: ${backgroundBottom};
+	width: 100%;
+	min-height: 100px;
+	p,
+	a,
+	li {
+		color: ${text.ctaPrimary};
+		${headline.xxxsmall()};
+		line-height: 135%;
+		margin-bottom: ${space[2]}px;
+	}
+
+	li::before {
+		content: "";
+		background-color: ${accentColour};
+		width: 0.75rem;
+		height: 0.75rem;
+		border-radius: 0.5rem;
+		display: inline-block;
+		vertical-align: middle;
+		margin-right: 0.5rem;
+	}
+
+	li {
+		padding-left: 1.25rem;
+		text-indent: -1.25rem;
+	}
+	a {
+		text-decoration: none;
+		padding-bottom: 0.005rem;
+		border-bottom: 1px solid ${accentColour};
+	}
+
+	${from.mobileLandscape} {
+		padding: ${space[1]}px ${space[5]}px;
+	}
+
+	${from.desktop} {
+		padding: ${space[1]}px 0;
+	}
+`;
+
+const bylineStyles: SerializedStyles = css`
+	margin: ${space[5]}px 0 ${space[3]}px;
+`;
+
+const leftColStyles: SerializedStyles = css`
+	display: inline-block;
+	height: 100%;
+	width: 220px;
+	padding: 0 ${space[5]}px;
+
+	${until.desktop} {
+		display: none;
+	}
+`;
+
+const containerStyles: SerializedStyles = css`
+	width: 100%;
+
+	${from.desktop} {
+		max-width: 700px;
+	}
+`;
+
+const updatedStyles: SerializedStyles = css`
+	color: ${text.ctaPrimary};
+	${textSans.xxsmall()};
+`;
+
+// The following pulsing dot implementation is a placeholder for the PulsingDot component, which
+// we will ideally lift up from dcr to common-rendering
+const allowsAnimation = "@media (prefers-reduced-motion: no-preference)";
+
+const livePulse = keyframes`
+    0% {opacity: 1;}
+    10% {opacity: .25;}
+    40% {opacity: 1;}
+    100% {opacity: 1;}
+`;
+
+const animation: SerializedStyles = css`
+	${allowsAnimation} {
+		animation: ${livePulse} 1s infinite;
+	}
+`;
+
+const dotStyles: SerializedStyles = css`
+	:before {
+		content: "";
+		background-color: ${text.ctaPrimary};
+		width: 0.55rem;
+		height: 0.55rem;
+		border-radius: 0.5rem;
+		display: inline-block;
+		vertical-align: middle;
+		margin-right: 0.25rem;
+	}
+`;
+
+// End of temporary pulsing dot implementation
+
+const Header: FC<HeaderProps> = ({
+	theme,
+	title,
+	credit,
+	standfirst,
+	byline,
+}) => {
+	const [themeColours] = useState<Colours>(() => getThemeColours(theme));
```

Comment:
This is just a question as I'm not super experienced in react. But I'm just wondeing why we need a state for themeColors? can it not be a function that would get the colour when the component is rendered?